paymentsheet/src/main/java/com/stripe/android/paymentsheet/PaymentSheetContractV2.kt

```diff
@@ -0,0 +1,72 @@
+package com.stripe.android.paymentsheet
+
+import android.app.Activity
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import androidx.activity.result.contract.ActivityResultContract
+import androidx.annotation.ColorInt
+import androidx.annotation.VisibleForTesting
+import androidx.core.os.bundleOf
+import com.stripe.android.core.injection.DUMMY_INJECTOR_KEY
+import com.stripe.android.core.injection.InjectorKey
+import com.stripe.android.view.ActivityStarter
+import kotlinx.parcelize.Parcelize
+
+internal class PaymentSheetContractV2 :
```

Comment:
Why do we need this version 2 of `PaymentSheetContract`? Unfortunately, `PaymentSheetContract` was exposed publicly, which makes it impossible to replace `clientSecret` with `initializationMode` in its constructor without affecting the auto-generated methods of the data class.

The ultimate resolution is this: We add a `rememberPaymentSheet()` method for Compose [here](https://github.com/stripe/stripe-android/pull/6230). Instantiating a `PaymentSheet` in Compose [seems to have been the driving force](https://github.com/stripe/stripe-android/issues/3803#issuecomment-874684636) behind exposing `PaymentSheetContract`, and by offering this improved integration form, we can deprecate and eventually restrict `PaymentSheetContract` again, at which point we can combine it and `V2`.